File: view/view/meubanco/src/app/DTO/cliente.ts
export class Cliente {
  id!: number;
  nome!: string;
  cpf!: string;
  email!: string;
  ativo!: boolean;
  observacoes!: string;

  constructor(obj?: { id: number; nome: string; cpf: string; email: string; ativo: boolean; observacoes: string; } | undefined) {
    if (obj) {
        this.id = obj.id;
        this.nome = obj.nome;
        this.cpf = obj.cpf;
        this.email = obj.email;
        this.ativo = obj.ativo;
        this.observacoes = obj.observacoes;

    }

}
}

File: model/src/main/java/com/indracompany/treinamento/model/entity/Cliente.java
package com.indracompany.treinamento.model.entity;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.validation.constraints.Email;

import com.fasterxml.jackson.annotation.JsonIgnore;

import org.hibernate.validator.constraints.br.CPF;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Table(name = "clientes")
@Data
@EqualsAndHashCode(callSuper = true)
public class Cliente extends GenericEntity<Long>{
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(length = 50)
	private String nome;
	
	@CPF
	@Column(length = 11)
	private String cpf;
	
	@Email
	private String email;
	
	private boolean ativo;
	
	private String observacoes;

	@JsonIgnore
	@OneToMany(mappedBy = "cliente", cascade = CascadeType.ALL, orphanRemoval = true)
	private Set<Conta> contas = new HashSet<>();
	
}
